presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -756,21 +761,28 @@ private TableFinishNode createTemporaryTableWrite(
                         partitioningScheme);
             }
 
-            TableWriterNode tableWriter = new TableWriterNode(
-                    idAllocator.getNextId(),
-                    writerSource,
-                    Optional.of(insertReference),
-                    variableAllocator.newVariable("partialrows", BIGINT),
-                    variableAllocator.newVariable("partialfragments", VARBINARY),
-                    variableAllocator.newVariable("partialtablecommitcontext", VARBINARY),
-                    outputs,
-                    outputColumnNames,
-                    Optional.of(partitioningScheme),
-                    Optional.empty(),
-                    Optional.empty());
+            String catalogName = tableHandle.getConnectorId().getCatalogName();
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+            TableStatisticsMetadata statisticsMetadata = metadata.getStatisticsCollectionMetadataForWrite(session, catalogName, tableMetadata.getMetadata());
+            StatisticsAggregationPlanner.TableStatisticAggregation statisticsResult = statisticsAggregationPlanner.createStatisticsAggregation(statisticsMetadata, columnNameToVariable, true);
```

Comment:
Statistic aggregation has a non trivial cost. Please guard it with a feature flag and disable it by default.